src/rxns/rxn_surface.F90

```diff
@@ -0,0 +1,295 @@
+
+  !> Constructor for surface reaction
+  function constructor() result(new_obj)
+
+    !> A new reaction instance
+    type(rxn_surface_t), pointer :: new_obj
+
+    allocate(new_obj)
+    new_obj%rxn_phase = AERO_RXN
+
+  end function constructor
+
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+  !> Initialize the reaction data, validating component data and loading
+  !! any required information into the condensed data arrays for use during
+  !! solving
+  subroutine initialize(this, chem_spec_data, aero_rep, n_cells)
+
+    !> Reaction data
+    class(rxn_surface_t), intent(inout) :: this
+    !> Chemical species data
+    type(chem_spec_data_t), intent(in) :: chem_spec_data
+    !> Aerosol representations
+    type(aero_rep_data_ptr), pointer, intent(in) :: aero_rep(:)
+    !> Number of grid cells to solve simultaneously
+    integer(kind=i_kind), intent(in) :: n_cells
+
+    type(property_t), pointer :: products, spec_props
+    character(len=:), allocatable :: key_name, reactant_name, product_name, &
+                                     phase_name, error_msg
+    integer :: i_spec, n_aero_jac_elem, n_aero_phase, i_phase, i_aero_rep, &
+               i_aero_id
+    integer, allocatable :: phase_ids(:)
+    real(kind=dp) :: temp_real
+
+    if (.not. associated(this%property_set)) call die_msg(244070915, &
+            "Missing property set needed to initialize surface reaction.")
+
+    key_name = "gas-phase reactant"
+    call assert_msg(807568174, &
+            this%property_set%get_string(key_name, reactant_name), &
+            "Missing gas-phase reactant name in surface reaction.")
+
+    key_name = "gas-phase products"
+    call assert_msg(285567904, &
+            this%property_set%get_property_t(key_name, products), &
+            "Missing gas-phase products for surface reaction.")
+
+    key_name = "aerosol phase"
+    call assert_msg(939211358, &
+            this%property_set%get_string(key_name, phase_name), &
+            "Missing aerosol phase in surface reaction.")
+
+    error_msg = " for surface reaction of gas-phase species '"// &
+                reactant_name//"' on aerosol phase '"//phase_name//"'"
+
+    call assert(362731302, associated(aero_rep))
+    call assert_msg(187310091, size(aero_rep) .gt. 0, &
+            "Missing aerosol representation"//error_msg)
+
+    ! Count the number of Jacobian elements needed in calculations of mass,
+    ! volume, etc. and the number of instances of the aerosol phase
+    n_aero_jac_elem = 0
+    n_aero_phase = 0
+    do i_aero_rep = 1, size(aero_rep)
+      phase_ids = aero_rep(i_aero_rep)%val%phase_ids(phase_name)
+      n_aero_phase = n_aero_phase + size(phase_ids)
+      do i_phase = 1, size(phase_ids)
+        n_aero_jac_elem = n_aero_jac_elem + &
+            aero_rep(i_aero_rep)%val%num_jac_elem(phase_ids(i_phase))
+      end do
+    end do
+
+    allocate(this%condensed_data_int(NUM_INT_PROP_ + 2 + 3 * products%size() + &
```

Comment:
Could you leave a comment here and for the following allocate statement on what the constants are (+2 + 3 *)?

Reply:
done